changelog/unreleased/pr-14424.toml

```diff
@@ -0,0 +1,24 @@
+type = "added"
+message = "Introduces a new index rotation strategy combining features of size-based and time-based strategies."
+
+issues = ["graylog-plugin-enterprise#4447"]
+pulls = ["14424", "graylog-plugin-enterprise#4572"]
+
+details.user = """
+The new index rotation strategy aims to make it easier to achieve good resource usage,
+i.e. shards that are neither too small nor too big, while avoiding excessive number
+of shards. This has been difficult to date, especially when faced with variable ingest
+rate.
+"""
+
+details.ops = """
+`time_size_optimizing` is the new default rotation strategy, as it provides the best results
+with a minimum of user input. The default settings are based on ES/OS sharding recommendations and generally
+do not need to be changed.
+The following configuration file settings are available, if you wish to modify the behavior
+for your requirements:
+`time_size_optimizing_rotation_min_shard_size`: index should be at least this big before rotating.
+`time_size_optimizing_rotation_max_shard_size`: rotate if index exceeds this size limit.
+`time_size_optimizing_rotation_min_lifetime`: index should be at least this old before rotating.
+`time_size_optimizing_rotation_max_lifetime`: rotate if index exceeds this age limit.
```

Comment:
I think the two last ones affect the retention, not the rotation, so the current naming can be misleading. 

As we write in some other places: 

> "time-size-optimizing" **tries to rotate daily**, while focussing on optimal sized shards. 

So as a user I would wonder if it still tries to rotate daily when `time_size_optimizing_rotation_min_lifetime` is set to e.g. 30 days.

Reply:
> I think the two last ones affect the retention, not the rotation, so the current naming can be misleading.
yes and no. 
The rotation needs to know the time range for the retention, so it knows for how long it can keep an index open.
While it would be more correct to put this setting into the retention, it is in fact implemented in the rotation strategy, because otherwise we would have to touch each existing retention.

So from a code perspective the rotation name is better.
From a user perspective, maybe not. OTOH, this defines only the default, which again can now be
tweaked in the system/configurations UI. So I think most users won't be using this setting. (cloud will be though)